thread_pool.h

```diff
@@ -25,35 +30,97 @@ class ThreadPool {
     unsigned num_active_threads = 0;
     //FIXME temporary impl
     std::vector<std::thread> threads;
+    bool stoping = false;
+    std::condition_variable event_var;
+    std::mutex event_mutex;
+    std::queue <std::function<void()>>  task_queue;
 
 public:
-    explicit ThreadPool(ThreadPool::Config config = {}) {} //TODO implement
+    explicit ThreadPool(ThreadPool::Config config = {}) {
+        start();
+    } //TODO implement
 
     ~ThreadPool() {
-        //FIXME temporary impl
-        for (auto& th : threads) {
-            th.join();
-        }
+        stop();
         //close active connections
         //reject pending connections
         //send shutdown signal
         //wait for threads destruction
     }
 
-    bool submit_job(std::function<void()>&& job) {
+    auto submit_job(std::function<void()>&& job)->std::future<decltype(job())> 
```

Comment:
This is a good method signature but leave some space before and after the arrow ("..)->s..." => "..) -> s...")